### PR TITLE
Adding Option for users to use Pulumi 

### PR DESCRIPTION
**Pull request by Marcus-Hines**
Adding option for users to use Pulumi instead of Terraform. Terraform will be the default. Dockerfile has been updated to reference the new shared image. This image has been created with Pulumi/Go and Terraform, and the image is in MCR

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM gambtho/azurecli_terraform:latest
```
Comment (gambtho) [inline Dockerfile:1]: nice :)

File: action.yml [hunk 2]
```diff
@@ -40,9 +40,17 @@ inputs:
...
     default: 'create'
+  USE_PULUMI:
+    description: 'use pulumi instead of terraform'
+    required: false
+    default: 'false'
   CREATE_ACR:
     description: 'create ACR with cluster name'
     default: 'false'
+  PULUMI_ACCESS_TOKEN:
+    description: 'create ACR with cluster name'
```
Comment (gambtho) [inline action.yml]: is the description and default correct for this?

File: action.yml [hunk 3]
```diff
@@ -40,9 +40,17 @@ inputs:
     description: 'create or destroy'
     required: false
     default: 'create'
+  USE_PULUMI:
```
Comment (gambtho) [inline action.yml]: maybe make this a string instead of a bool -- and they can pick the IaC tool?

File: action.yml (hunk 2 above)
Comment (Marcus-Hines) [reply action.yml]: no it's wrong. I'll update

File: action.yml (hunk 3 above)
Comment (Marcus-Hines) [reply action.yml]: Great suggestion. I updated it to be IAC_TYPE, and listed the current possible choices in the description

**Review by gambtho — APPROVED** [review]

File: action.yml [hunk 4]
```diff
@@ -40,9 +40,17 @@ inputs:
...
+  IAC_TYPE:
+    description: 'IAC tool, Possible choices: [ pulumi, terraform ]'
+    required: false
+    default: 'terraform'
   CREATE_ACR:
     description: 'create ACR with cluster name'
     default: 'false'
+  PULUMI_ACCESS_TOKEN:
+    description: 'Token that allows this action to access Pulumi'
+    default: 'false'
```
Comment (OliverMKing) [inline action.yml]: should this be default or required?

**Review by OliverMKing — APPROVED** [review]

File: action.yml (hunk 4 above)
Comment (Marcus-Hines) [reply action.yml]: lol required, great catch

**Pull request merged**